rules/mysql-to-mysqli/src/Rector/FuncCall/MysqlQueryMysqlErrorWithLinkRector.php

```diff
@@ -82,44 +98,64 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Node
     {
-        $connectionVariable = $this->findConnectionVariable($node);
-
-        // no connection? → unable to refactor
-        if ($connectionVariable === null) {
-            return null;
-        }
-
         foreach (self::FUNCTION_RENAME_MAP as $oldFunction => $newFunction) {
             if (! $this->isName($node, $oldFunction)) {
                 continue;
             }
 
+            if (
+                count($node->args) === 0
+                || ! $this->isProbablyMysql($node->args[0]->value)
```

Comment:
An alternative approach to this would be to partition the handled functions by the new function arity, then add a link/connection parameter whenever the number of args applied is less than what's required.

That may handle the case of externally initialised link arguments better (where PHPStan doesn't seem to ascertain the type correctly, and assumes that the link is not a resource or `mysqli` object (leading to too many parameters being provided (similar to the pre-existing code))).